File: src/main/java/com/hiddewieringa/elevator/domain/elevator/query/ElevatorQueryHandler.java
package com.hiddewieringa.elevator.domain.elevator.query;

import com.hiddewieringa.elevator.domain.elevator.model.ElevatorAggregate;
import org.axonframework.commandhandling.model.Aggregate;
import org.axonframework.commandhandling.model.Repository;
import org.axonframework.queryhandling.QueryHandler;
import org.axonframework.spring.config.AnnotationDriven;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ElevatorQueryHandler {
//
//    private final Repository<ElevatorAggregate> elevatorAggregateRepository;
//
//    @Autowired
//    public ElevatorQueryHandler(Repository<ElevatorAggregate> elevatorAggregateRepository) {
//        this.elevatorAggregateRepository = elevatorAggregateRepository;
//    }
//
//    @QueryHandler
//    public FloorResult queryFloor(QueryFloor query) {
//        Aggregate<ElevatorAggregate> q= elevatorAggregateRepository.load(query.getElevatorId().toString());
//        return new FloorResult(q.invoke(ElevatorAggregate::getFloor));
//    }
}

File: src/main/java/com/hiddewieringa/elevator/domain/elevator/query/ElevatorQueryHandler.kt
package com.hiddewieringa.elevator.domain.elevator.query

import com.hiddewieringa.elevator.domain.elevator.ElevatorCreated
import com.hiddewieringa.elevator.domain.elevator.model.ElevatorId
import org.axonframework.eventhandling.EventHandler
import org.springframework.stereotype.Service
import java.util.*

@Service
open class ElevatorQueryHandler {

    private val activeElevators = ArrayList<ElevatorId>()

    @EventHandler
    fun create(event: ElevatorCreated) {
        activeElevators.add(event.elevatorId)
    }

    open fun active(): List<ElevatorId> {
        return Collections.unmodifiableList(activeElevators)
    }
}
